range_slider_test.dart diagnostics property tests

sladecapasso opened the pull request:
## Description

* This PR is to add DiagnosticsProperties tests to Material Range Slider, to address issue #40277

## Related Issues

* #40277

## Tests

I added the following tests:

* DiagnosticsProperties

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

sladecapasso committed:
Updates to slider_test.dart

Fixed error that required onChanged.

sladecapasso committed:
Updated slider.dart and slider_test.dart debugFillProperties

Updated slider.dart debugFillProperties to include all properties and slider_test debugFillProperties to test all properties

sladecapasso committed:
Updated Slider onChanged property

sladecapasso committed:
Updated debugFillProperties to ignore testing methods that are null in test

Per feedback from Hans/Anthony on Flutter Material

sladecapasso committed:
Updated spacing in debugFillProperties

Per feedback from Hans

sladecapasso committed:
Updated range_slider.dart and range_slider_test.dart to have DiagnosticProperties tests for all properties

sladecapasso committed:
Merge branch 'master' of https://github.com/flutter/flutter into range_slider_test.dart-DiagnosticsProperty-tests

johnsonmh commented on packages/flutter/lib/src/material/range_slider.dart:
```diff
@@ -337,6 +337,24 @@ class RangeSlider extends StatefulWidget {
+    properties.add(DoubleProperty('min', min));
+    properties.add(DoubleProperty('max', max));
+    properties.add(IntProperty('divisions', divisions));
+    properties.add(StringProperty('label1', labels.start));
```
Although `values` can never be null, it looks like `labels` can be null. If `labels` is null and someone calls `debugFillProperties`, it will through an error.

To protect against this, change this to `labels?.start` and `labels?.end`.

johnsonmh commented on packages/flutter/test/material/range_slider_test.dart:
```diff
@@ -1533,5 +1533,41 @@ void main() {
   });
+
+  testWidgets(
+      'Range Slider implements debugFillProperties', (WidgetTester tester) async {
```
This can all be one line.

```dart
testWidgets('Range Slider implements debugFillProperties', (WidgetTester tester) async {
```

johnsonmh reviewed (approved):
LGTM

johnsonmh commented on packages/flutter/test/material/range_slider_test.dart:
```diff
@@ -1533,5 +1533,41 @@ void main() {
+      activeColor: Colors.blue,
+      divisions: 4,
+      inactiveColor: Colors.grey,
+      labels: RangeLabels('lowerValue', 'upperValue'),
```
```suggestion
      labels: const RangeLabels('lowerValue', 'upperValue'),
```

johnsonmh commented on packages/flutter/test/material/range_slider_test.dart:
```diff
@@ -1533,5 +1533,41 @@ void main() {
+      onChangeEnd: null,
+      onChangeStart: null,
+      semanticFormatterCallback: null,
+      values: RangeValues(25.0, 75.0),
```
```suggestion
      values: const RangeValues(25.0, 75.0),
```

sladecapasso committed:
Updated range_slider.dart and range_slider_test.dart per feedback

Preventing null labels from creating an error.

Added made RangeLabels and RangeValues constants in instantiation of Range Slider.

Fixed spacing.

clocksmith commented on packages/flutter/lib/src/material/range_slider.dart:
```diff
@@ -337,6 +337,24 @@ class RangeSlider extends StatefulWidget {
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DoubleProperty('value1', values.start));
```
change to start and end

clocksmith commented on packages/flutter/lib/src/material/range_slider.dart:
```diff
@@ -337,6 +337,24 @@ class RangeSlider extends StatefulWidget {
+    properties.add(DoubleProperty('min', min));
+    properties.add(DoubleProperty('max', max));
+    properties.add(IntProperty('divisions', divisions));
+    properties.add(StringProperty('label1', labels?.start));
```
change to start and end

sladecapasso committed:
Updates to start and end labels in diagnostic property tests

Updated per feedback from @clocksmith

rami-a reviewed (approved):
LGTM!

HansMuller reviewed (approved):
LGTM
